icu4j/main/tests/core/src/com/ibm/icu/dev/test/util/CurrencyTest.java

```diff
@@ -357,7 +357,7 @@ public void TestCurrencyDisplayNames() {
         assertNull("de_FOO_plural_foo", cdn.getPluralName("FOO", "foo"));
 
         // unknown locale with no substitute
-        cdn = CurrencyDisplayNames.getInstance(ULocale.forLanguageTag("zz-Gggg-YY"), true);
+        cdn = CurrencyDisplayNames.getInstance(ULocale.forLanguageTag("zz-YY"), true);
```

Comment:
I had to take the script code out of both instances of "zz-Gggg-YY" in this file-- thanks to the new default-script logic, these would fall straight back to "root", rather than to the system default locale, so I fixed it by removing the script code.

Reply:
Ok

Reply:
Is this right? It seems like zz-Gggg should skip zz and *go to the next thing in the fallback chain*, which for the "normal" locale trees would be the default locale.

Reply:
The way I have this coded right now, no.  If the parent locale table says "root", I go to root, not to the default locale. Do we have consensus that this is the wrong thing to do, and I should go through the default locale in this case?

Reply:
Never mind.  I changed this, and it didn't hurt anything, so I'm just going to make this change.

Reply:
I was actually questioning myself on this one. The current behavior for when we have the data, like `%%Parent{"root"}`, is to go straight to root, and not via the default locale, right? So what you did was equivalent, just extending it to all locales for which CLDR has data, or where the CLDR algorithm holds (skip the language if it implies a different script). Sorry if I am causing you to flip-flop.

Does anyone else want to chime in here?

Reply:
What I originally did is what you're describing here, and it broke those tests.  To be honest, I'm not sure what the Right Answer is supposed to be in this situation, but if the `%%Parent` directives in the resource bundles themselves send us straight to `root` , that's an argument for what I originally had (and changing the tests in some way).  Unless we tend to believe the `%%Parent` directives were doing the wrong thing...

I think you're right-- I'd like other people to chime in and for us to come to some kind of consensus as to what the proper behavior here is.  The change I made is easy enough to undo, but I don't want to have to redo it again if the discussion goes that way.  Is carrying on the discussion here good enough, or should I sent out an email?

Reply:
> For resource bundle lookup we insert the default locale into the chain. That's basically saying that if we fail to find your locale, we'll try the system's default.

We really have two related algorithms here:
- Looking for the first (most specific) bundle for a given locale ID. If we find nothing at all, then we go through the default locale (for display-oriented services).
- Once we have a bundle, also load the parent locale bundle and its parent etc., down to root. This never goes through the default locale.

In ICU so far, we get the parent locale information from inside the bundles, which means that the two algorithms visit different parents when we are missing files. When we find zh_Hant, `%%Parent` sends us to root (and we have *not* been going to the default locale). When zh_Hant is missing, we truncate to zh.

With Rich's change, we should be able to remove the `%%Parent` entries, because they will be redundant with the CLDR parent locale data. I think this should also mean that we now build parent-locale chains (down from a found bundle) consistently like we look for the first (most specific) available bundle.

So if we find zh_Hant, then we skip zh, and only root is the parent in the runtime bundle chain.
If we don't find zh_Hant, then we also skip zh, and... I think load root, not default?
And if so, then if we don't find zz_Gggg, then we skip zz and also load root, not default?

Reply:
> https://unicode.org/reports/tr35/#Bundle_vs_Item_Lookup That's also different than inherited item lookup, where we are pulling in a particular item, and looking in the parent locale. For that, we don't want to involve the default locale at all, otherwise we can get a severe ransom-note effect.

The point is, for services like collation and break iteration, we don't go through the default locale *even in bundle lookup*.

Reply:
>For resource bundle lookup we insert the default locale into the chain. That's basically saying that if we fail to find your locale, we'll try the system's default.

I read this to mean the logic I have right now is correct, and when the parent locale table says "root" or we have a non-default script, we want to fall back through the default locale, instead of directly to root.  @macchiati and @markusicu , do you agree?

>With Rich's change, we should be able to remove the %%Parent entries, because they will be redundant with the CLDR parent locale data. I think this should also mean that we now build parent-locale chains (down from a found bundle) consistently like we look for the first (most specific) available bundle.

I don't actually think this is true, without more work.  My change enables us to get rid of resource bundles that have nothing in them but a `%%Parent` resource, but if a resource bundle has a `%%Parent` resource _and other resources_, we'd need extra code to make _inherited item lookup_ use the parent locale table instead of the `%%Parent` resources.  I think this is a good idea, but I also think this PR is getting really big, and we should handle that as a separate ticket after I land this one, maybe in a future version of ICU.

Reply:
Thanks for the explanation. I had thought that the goal was to put the new parent-locale data to work in both bundle lookup and bundle-parent-chain building. I now understand that it's only the former. So, good.